### PR TITLE
Handle obsolete messages in PO

### LINKED ISSUE #9: Support of Obsolete Messages
Hello,

have you planned to support obsolete messages in your po parser?

Example of obsolete messages in a po file:

```
#~ msgid "Obsolete message"
#~ msgstr "Message obsolète"
```

regards,

### PR DESCRIPTION
**Pull request by KapitanOczywisty**
Obsolete translations are kept in optional `obsolete` property:
```jsonc
{
  //...
  obsolete: {
    '': {
      'Obsolete message': {
        msgid: 'Obsolete message',
        comments: { translator: 'Obsolete comment' },
        msgstr: [
          'Message obsolete.\nLines are properly handled, even for obsolete translations.'
        ]
      }
    }
  }
}
```
and properly compiled back into `po` file:
```po
# Obsolete comment
#~ msgid "Obsolete message"
#~ msgstr ""
#~ "Message obsolete.\n"
#~ "Lines are properly handled, even for obsolete translations."
```

Closes #9

### CODE REVIEW

**Comment by smhg** [comment]
This is great! Thank you!
One thing: can you add/update docblocks for the methods you've added/changed?

**Comment by KapitanOczywisty** [comment]
@smhg something like that?

File: lib/pocompiler.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ Compiler.prototype._drawComments = function (comments) {
  * @param {Object} [override] Properties of this object will override `block` properties
```
Comment (smhg) [inline lib/pocompiler.js:107]: Please add the obsolete parameter here too.

**Comment by KapitanOczywisty** [comment]
I forgot about that, fixed.

**Comment by rankida** [comment]
Is it intended that now all new comments with a tilde `~` in it fail parsing?
I expected only those lines _starting_ with `#~` would be considered as obsolete comments but it seems if the first comment line has a `~` anywhere in the comment it is picked up, and so expects a regular comment on the proceeding line.

For example the below PO file now returns the error message `Error parsing PO data: Invalid key name "obsolete" at line 13. This can be caused by an unescaped quote character in a msgid or msgstr value.`

```PO
msgid ""
msgstr ""
"Project-Id-Version: 10.0.0\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Plural-Forms: nplurals=2; plural=(n != 1)\n"
""

# Text: this is NOT ~ obsolete
msgid "test
msgstr "regular text"
```

File: lib/poparser.js [hunk 2]
```diff
@@ -161,13 +164,19 @@ Parser.prototype._lexer = function (chunk) {
...
+            this._node.obsolete = true;
+          }
           this._lex.push(this._node);
           this._state = this.states.key;
         }
         break;
       case this.states.comments:
         if (chr === '\n') {
           this._state = this.states.none;
+        } else if (chr === '~') {
```
Comment (KapitanOczywisty) [inline lib/poparser.js:177]:
```diff
- } else if (chr === '~') {
+ } else if (chr === '~' && this._node.value === '') {
```

@smhg can you patch this?

**Comment by smhg** [comment]
Thank you for reporting and patching this. This fix is released in version 5.1.1.
Please do me a favor and submit a separate issue and PR next time :)

**Comment by rankida** [comment]
Thanks so much 🙇 

Sorry for not following the process. Really appreciate the response. Cheers!

**Pull request merged**